packages/react/src/analytics/integrations/shared/GoogleConsentMode/GoogleConsentMode.ts

```diff
@@ -0,0 +1,93 @@
+import { type ConsentData } from '@farfetch/blackout-analytics';
+import { type GoogleConsentModeMappings } from './types.js';
+
+/**
+ * GoogleConsentMode handles with Google Consent Mode v2.
+ */
+export class GoogleConsentMode {
+  private dataLayer!: string; // Stores different data layer names
+  private consentCategoriesMappings?: GoogleConsentModeMappings; // Stores default or customized consent category mappings
+
+  constructor(
+    dataLayer: string,
+    consent: ConsentData | null,
+    consentCategoriesMapping?: GoogleConsentModeMappings,
+  ) {
+    this.dataLayer = dataLayer;
+    this.consentCategoriesMappings = consentCategoriesMapping;
+
+    this.write('consent', 'default', this.getConsentValues(consent));
+  }
+
+  /**
+   * Retrieves consent values for each category based on provided consents.
+   *
+   * @param consent - The consent data.
+   *
+   * @returns Consent values for each category.
+   */
+  private getConsentValues(
+    consent: ConsentData | null,
+  ): Record<string, Array<string> | string | number | undefined> {
+    // Dealing with null or undefined consent values
+    const consentList = consent || {};
+
+    if (this.consentCategoriesMappings) {
+      return Object.keys(this.consentCategoriesMappings).reduce(
+        (result, consentKey) => ({
+          ...result,
+          [consentKey]: this.consentCategoriesMappings?.[consentKey]?.every(
+            consent => consentList[consent],
+          )
+            ? 'granted'
+            : 'deny',
+        }),
+        {},
+      );
+    }
+
+    // Return default google consent values as deny all
+    return {
+      ad_storage: 'deny',
+      ad_user_data: 'deny',
+      ad_personalization: 'deny',
+      analytics_storage: 'deny',
+    };
+  }
+
+  /**
+   * Update consent.
+   *
+   * @param consent - The consent data to be set.
+   */
+  updateConsent(consent: ConsentData) {
+    this.write('consent', 'update', this.getConsentValues(consent));
+  }
+
+  /**
+   * Handles consent by updating the data layer with consent information.
+   *
```

Comment:
Add a commentary saying why the arguments are being ignored so other developers will know what is the reason behind it.

Reply:
Fixed